app/models/theme.rb

```diff
@@ -435,25 +435,30 @@ def translation_override_hash
   end
 
   def generate_metadata_hash
-    {
-      name: name,
-      about_url: remote_theme&.about_url,
-      license_url: remote_theme&.license_url,
-      component: component,
-      assets: {}.tap do |hash|
+    {}.tap do |meta|
+      meta[:name] = name
+      meta[:component] = component
+
+      RemoteTheme::METADATA_PROPERTIES.each do |property|
```

Comment:
Could you use `slice` here?  `meta = remote_theme.slice(RemoteTheme::METADATA_PROPERTIES)`

Reply:
We want the output to contain all the keys, even if the values are `nil`, AND even if there is no `remote_theme`. 

This works for missing values:
```
meta.merge! remote_theme.slice(RemoteTheme::METADATA_PROPERTIES)
```

But to also account for a missing remote_theme I'd need:

```
meta.merge! (remote_theme || {})&.slice(RemoteTheme::METADATA_PROPERTIES) ||
            Hash[RemoteTheme::METADATA_PROPERTIES.map { |x| [x, nil] }]
```

I think the original looping method is probably easier to read and understand